### PR TITLE
Add stubs for Relay

### PR DESCRIPTION
**Pull request by tillkruss**
This PR adds support for [Relay](https://github.com/cachewerk/relay), a next-gen Redis client written in C by the makers of PhpRedis and Predis. It’s built as a drop-in replacement for PhpRedis with a backwards compatible interface for easy adoption. Relay is significantly faster than existing clients by leveraging Redis 6's client-side-caching.

While Relay is still on 0.x (pending the addition of cluster support in a few weeks for a 1.0 tag), it’s interface is stable and it’s heavily used in production deployments.

The stub files are based on: https://builds.r2.relay.so/dev/relay.stub.php

### CODE REVIEW

**Comment by tillkruss** [comment]
@isfedorov: Could you give me pointers what's needed to get these stubs merged?

**Comment by isfedorov** [comment]
@tillkruss do you have any statistics about the current usage of the Relay extension? The extension looks like a very early build (based on 0.X version, amount of commits, forks, stars, etc) so sorry, but at the moment I don't see any guarantee that the work on the extension will continue. Unfortunately, we can't bundle stubs for every extension and would like to align with stubs for the extensions that are going to live for a while and are used by a significant amount of people. So could you please provide some data on usage the extension (monthly installs, summary installs etc)?

**Comment by tillkruss** [comment]
@isfedorov You're absolutely right, it's early days and we've only soft launched, but 3 people are working full time on Relay and just over 10k active (production) installations right now. Those are early adopters in the WordPress hosting space.

It would be great to get this merged, since Laravel and Symfony both are adding support for it.

Are those numbers sufficient, or do you have any specific usage requirements?

**Review by isfedorov — CHANGES_REQUESTED** [review]
Please remove `@return` tag from constructors as they look completely redundant. 
Please replace `@alias` tags with the supported tag. 
Please update stubs map https://github.com/JetBrains/phpstorm-stubs#how-to-update-stub-map
Please add `relay` to `\StubTests\TestData\Providers\Stubs\PhpCoreStubsProvider::$StubDirectoryMap` to `OTHERS` array

**Comment by isfedorov** [comment]
@tillkruss Ok, lets bundle them. But before that the PR requires some changes (please see my notes)

File: relay/Relay.php [hunk 1]
```diff
@@ -0,0 +1,3171 @@
...
+
+    /**
+     * Establishes a new connection to Redis, or re-uses already opened connection.
+     *
+     * @param  string  $host
+     * @param  int  $port
+     * @param  float  $connect_timeout
+     * @param  float  $command_timeout
+     * @param  array  $context
+     * @return Relay
```
[the remark this answers is not in the document]
Comment (szepeviktor) [reply relay/Relay.php:384]: Constructor return types should not be written anywhere.

[the remark this answers is not in the document]
Comment (tillkruss) [reply relay/Relay.php:384]: Thanks, will remove.

File: relay/Relay.php [hunk 2]
```diff
@@ -0,0 +1,3171 @@
...
+    /**
+     * Returns the read timeout.
+     *
+     * @return float|false
+     */
+    #[\Relay\Attributes\Local]
+    public function getReadTimeout(): float|false {}
+
+    /**
+     * @alias Relay\Relay::getReadTimeout
```
[the remark this answers is not in the document]
Comment (tillkruss) [reply relay/Relay.php:574]: I see. PHP src actually uses the `@alias` tag for linking methods. Will replace here.

**Comment by tillkruss** [comment]
@isfedorov: All done. Happy to do the `LanguageLevelTypeAware` stuff if needed.

**Comment by isfedorov** [comment]
@tillkruss Ok, so the only problem left is cs-fixer problems. Please run cs-fixer as mentioned at https://github.com/JetBrains/phpstorm-stubs/blob/master/CONTRIBUTING.md#code-style

**Comment by tillkruss** [comment]
Done 👍

**Review by szepeviktor — APPROVED** [review]

**Pull request merged**